### PR TITLE
feat: manage subscription list from unsubscribe page

### LINKED ISSUE #4: feat: add list of email types on unsubscription page
# Issues

Current unsubscribe page auto send a request to unsubscribe on page load

# Changes

We should remove the auto unsubscribe send request on page load, and trigger it only when the user click on a UNSUBSCRIBE button

Also add a form, so user can select the email types he wants to unsubscribe from (list will be retrieve from envelop API)

e.g.:

Please select the email you want to unsubscribe from 

[] Proposal creation (<- checkbox)
[] Proposal closing
[] Weekly report 

Select all options (<- js event to select all checkboxes)

[UNSUBSCRIBE] (<- submit button)

### PR DESCRIPTION
**Pull request by wa0x6e**
## 🧿 Current issues / What's wrong ?

There is currently no page to allow user to update his subscription preferences, only a unsubscribe page.

## 💊 Fixes / Solution

Fix #4

Add a new form to the unsubscribe page, inviting user to `update` and choose what emails to receive, in addition to `unsubscribe`, as a last resort to retain him.

## 🚧 Changes

- Add a form to the unsubscribe page, before the UNSUBSCRIBE button, to invite user to update his subscription preferences, instead of unsubscribing

Has to be tested with this PR of envelop backend https://github.com/snapshot-labs/envelop/pull/55

### CODE REVIEW

**Comment by wa0x6e** [comment]
Obsolete feature

File: src/views/Home.vue [hunk 1]
```diff
@@ -1,3 +1,9 @@
+<script setup lang="ts">
+import BrandLogo from '@/components/BrandLogo.vue';
+</script>
+
 <template>
-  <main></main>
+  <div :class="['max-w-screen-xs', 'xs:w-[420px]', 'p-5', 'py-6', { 'text-center': true }]">
```
Comment (Todmy) [inline src/views/Home.vue]:
```suggestion
  <div class='max-w-screen-xs xs:w-[420px] p-5 py-6 text-center">
```

File: src/views/Verify.vue [hunk 2]
```diff
@@ -19,60 +19,58 @@ const route = useRoute();
 
 useSeoMeta({
   title: 'Verify email',
-  description: 'Verify your email to confirm your subscription to Snapshot weekly summary'
+  description: 'Verify your email to confirm your subscription to Snapshot mailing list'
 });
 
-function verify() {
+async function verify() {
```
Comment (Todmy) [inline src/views/Verify.vue:26]: we can [use](https://vueuse.org/core/useFetch/#usefetch) `useFetch` instead

**Review by Todmy — CHANGES_REQUESTED** [review]

**Comment by Todmy** [comment]
also, subscriptions on the `unsubscribe page` subscriptions are not properly defined from the user profile(DB). If a user subscribed to one type of email, the page displays that he/she subscribed to all types.

File: src/views/Verify.vue (hunk 2 above)
Comment (wa0x6e) [reply src/views/Verify.vue:25]: Weird, should already be present: https://github.com/snapshot-labs/envelop-ui/blob/unsubscribe-message/src/views/Verify.vue#LL25C1-L25C1

**Comment by wa0x6e** [comment]
> also, subscriptions on the unsubscribe page subscriptions are not properly defined from the user profile(DB). If a user subscribed to one type of email, the page displays that he/she subscribed to all types.

Seems like loading the user's existing subscription is not in the scope of this PR. And there are no endpoint on the BE that can return such list, only available endpoint now only return a list by address, and not by email.

Should we go with this PR as is, and implement that feature in another PR, or do it inside this one?

**Comment by Todmy** [comment]
> > also, subscriptions on the unsubscribe page subscriptions are not properly defined from the user profile(DB). If a user subscribed to one type of email, the page displays that he/she subscribed to all types.
> 
> Seems like loading the user's existing subscription is not in the scope of this PR. And there are no endpoint on the BE that can return such list, only available endpoint now only return a list by address, and not by email.
> 
> Should we go with this PR as is, and implement that feature in another PR, or do it inside this one?

but we have [it](https://github.com/snapshot-labs/envelop/blob/main/src/rpc.ts#L76)

**Comment by wa0x6e** [comment]
> > also, subscriptions on the unsubscribe page subscriptions are not properly defined from the user profile(DB). If a user subscribed to one type of email, the page displays that he/she subscribed to all types.
> 
> Seems like loading the user's existing subscription is not in the scope of this PR. And there are no endpoint on the BE that can return such list, only available endpoint now only return a list by address, and not by email.
> 
> Should we go with this PR as is, and implement that feature in another PR, or do it inside this one?

This one is for subscriptions by wallet address. The page from envelop-ui requires a list by email address, and it will affects all wallet addresses related to this email.

File: src/views/Verify.vue (hunk 2 above)
Comment (wa0x6e) [reply src/views/Verify.vue:26]: We can, but is it providing additional value from the native `fetch` ? The coding guideline is advocating to use as less dependencies as possible.

**Comment by wa0x6e** [comment]
> This one is for subscriptions by wallet address. The page from envelop-ui requires a list by email address, and it will affects all wallet addresses related to this email.

I've opened #7 for this

File: src/views/Verify.vue (hunk 2 above)
Comment (wa0x6e) [reply src/views/Verify.vue:26]: Added

Comment (wa0x6e) [reply src/views/Verify.vue:26]: Only migrated the code in `verify`, `unsubscribe` has been migrated via #8

**Review by Todmy — APPROVED** [review]
tACK

**Pull request merged**